scripts/src/errors.ts

```diff
@@ -7,21 +41,54 @@ export type ErrorType =
 	| 'ServerError'
 	| 'FriendbotError'
 	| 'InvalidAddressError'
-	| 'TransactionFailedError'
 	| 'ChannelBusyError'
+	| 'TransactionFailedError'
 	| 'NetworkMismatchedError'
-	| 'InvalidDataError';
+	| 'InvalidDataError'
+	| 'BadRequestError'
+	| 'InternalError'
+	| 'NoAccountError'
+	| 'AccountExists'
+	| 'LowBalanceError'
+	| 'AccountNotActivatedError'
+	| 'HorizonError'
+	| 'ResourceNotFoundError';
 
 export interface KinSdkError extends Error {
 	readonly type: ErrorType;
 }
 
+export class HorizonError extends Error implements KinSdkError {
+	readonly type: ErrorType = 'HorizonError';
+
+	private readonly _resultTransactionCode?: string;
+	private readonly _resultOperationsCode?: string[];
+	readonly errorCode: number;
```

Comment:
make private, add getter?
wondering if we really need getters for simple properties